index.js

```diff
@@ -54,11 +55,23 @@ function decorateLayer(layer, keysToKeep, newProps) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
+    if (requiredKeys && requiredKeys.length) {
+        for (i = 0; i < requiredKeys.length; i++) {
+            var index = keys.indexOf(requiredKeys[i]);
+            if (index !== -1) requiredIndices.push(index);
+        }
+    }
+
     for (i = 0; i < layer.features.length; i++) {
         var feature = layer.features[i];
         var tags = feature.tags;
         feature.tags = [];
 
+        if (!hasAllKeys(tags, requiredIndices)) {
+            layer.features.splice(i--, 1);
+            continue;
+        }
```

Comment:
`splice` is very expensive. Can we avoid it? `layer.features = layer.features.filter(...)` would probably be much faster.